src/Utils.js

```diff
@@ -161,13 +162,14 @@ export function withSetStateAllowed(fn) {
   }
 }
 
-export function splitSelector(selector) {
-  return selector.split(/(?=\.|\[.*\])|(?=#|\[#.*\])/);
+export function getAst(selector) {
+  let value;
+  parser(ast => { value = ast; }).process(selector);
+  return value;
 }
 
-export function isSimpleSelector(selector) {
-  // any of these characters pretty much guarantee it's a complex selector
-  return !/[~\s:>]/.test(selector);
+export function isValidPropName(propName) {
+  return /^[$A-Z_][0-9A-Z_$-:]*$/i.test(propName);
```

Comment:
Rather than using `/i`, could we just specify `a-zA-Z`? I think it's equally readable but would be more efficient.